lib/Rex/Commands/Rsync.pm

```diff
@@ -192,11 +192,13 @@ sub sync {
   $cmd = join( " ", @rsync_cmd );
 
   if ( !$local_connection ) {
-    my $pass           = $auth->{password};
+    my $pass;
+    $pass = $auth->{password} if $auth;
     my @expect_options = ();
 
-    my $auth_type = $auth->{auth_type};
-    if ( $auth_type eq "try" ) {
+    my $auth_type;
+    $auth_type = $auth->{auth_type} if $auth;
+    if ( $auth && $auth_type eq "try" ) {
```

Comment:
Since we are touching the code anyway, and the logic is modified a bit, would it be a good time to make it simpler and/or more readable by grouping the similar statements together?

E.g.:

```
my ( $pass, $auth_type );
my @expect_options = ();

if ($auth) {
  $pass      = $auth->{password};
  $auth_type = $auth->{auth_type};
};
```

which might even opens up the possibility to use a single hash slice instead:

```
my ( $pass, $auth_type ) = @auth{qw(password auth_type)} if $auth;
my @expect_options       = ();
```